### PR TITLE
functors task 1 Vlad Yevka

### PR DESCRIPTION
**Pull request by yevka**
I have [1]    3086 segmentation fault  ./bin/unit_tests
 in test TEST(Generator, SemiFit). Reproduce only Mac Os X.

### CODE REVIEW

**Comment by yevka** [comment]
Bug was fixed by the @smay1613 .

File: functors task (1)/student/tests/unit_tests/src/maintest.cpp [hunk 1]
```diff
@@ -70,6 +70,7 @@ TEST(Generator, SemiFit)
         it = std::find(std::next(it),
```
Comment (smay1613) [inline functors task (1)/student/tests/unit_tests/src/maintest.cpp:70]: fixed, thanks

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -40,9 +52,19 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcoholDrink, NonAlcoholDrink nonAlcoholDrink)
     {
+        if (AlcoholDrink::Gin == alcoholDrink && NonAlcoholDrink::LimeJuice == nonAlcoholDrink) {
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:57]: consider switch

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -8,6 +8,16 @@
...
+    BeerBrand operator()() {
+        if (static_cast<int>(BeerBrand::Max) == current_beer + 1) {
+            current_beer = 1;
+        } else {
+            ++current_beer;
+        }
+        return static_cast<BeerBrand>(current_beer);
+    }
+private:
+    int current_beer = {0};
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:20]: You can remove "="

**Review by smay1613 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**